python/dgl/graphbolt/impl/uniform_negative_sampler.py

```diff
@@ -68,4 +69,15 @@ def _sample_with_etype(self, node_pairs, etype=None, use_seeds=False):
                 self.negative_ratio,
             )
         else:
-            raise NotImplementedError("Not implemented yet.")
+            return self.graph.sample_negative_seeds_uniform(
+                etype,
+                node_pairs,
+                self.negative_ratio,
+            )
+
+    def _construct_indexes(self, node_pairs):
```

Comment:
Instead of _construct_indexes, should this be part of _sample_with_etype's return?

Reply:
+1, add a separate function should scarify flexibility at cost